### ISSUE TITLE
Yabridge is crashing when running REAPER from launcher

### ISSUE THREAD

**Issue by aidanhadley** [robbert-vdh/yabridge#328]
### Thanks for giving yabridge a shot!

- [X] I read through both the troubleshooting and the known issues sections, and my issue wasn't listed there

### Problem description

Ubuntu 23.10, latest version of yabridge, wine-staging 9.4. When i run REAPER from the terminal, I'm able to load windows plugins (such as Meowsynth, one of my favorites) just fine. However, when I run it from the cockos-reaper.desktop launcher, it crashes. Logs and info are below.

### What did you expect to happen?

Reaper to not crash

### What actually happened?

Reaper crashed

### Operating system

Ubuntu 23.10

### How did you install yabridge?

Releases page

### yabridge version

5.1.0

### yabridgectl version

5.1.0

### Wine version

wine-staging 9.4

### Plugin

All Windows Plugins, but Meowsynth is the first I noticed

### Plugin type

both VST2 and VST3

### Plugin architecture

both 64-bit and 32-bit

### Host

REAPER 7.13

### Desktop environment or WM

GNOME 45

### Debug log

22:49:14 [MeowSynth-08Zfghor] library:       '/home/ldas/.local/share/yabridge/libyabridge-vst2.so'
22:49:14 [MeowSynth-08Zfghor] host:          '/home/ldas/.local/share/yabridge/yabridge-host-32.exe'
22:49:14 [MeowSynth-08Zfghor] plugin:        '/home/ldas/.wine/drive_c/Program Files/Steinberg/VstPlugins/MeowSynth.dll'
22:49:14 [MeowSynth-08Zfghor] plugin type:   'VST2'
22:49:14 [MeowSynth-08Zfghor] realtime:      'no'
22:49:14 [MeowSynth-08Zfghor] sockets:       '/run/user/1000/yabridge-MeowSynth-08Zfghor'
22:49:14 [MeowSynth-08Zfghor] wine prefix:   '/home/ldas/.wine'
22:49:14 [MeowSynth-08Zfghor] wine version:  '<NOT FOUND>'
22:49:14 [MeowSynth-08Zfghor] 
22:49:14 [MeowSynth-08Zfghor] config from:   '<defaults>'
22:49:14 [MeowSynth-08Zfghor] hosting mode:  'individually, 32-bit'
22:49:14 [MeowSynth-08Zfghor] other options: '<none>'
22:49:14 [MeowSynth-08Zfghor] 
22:49:14 [MeowSynth-08Zfghor] Enabled features:
22:49:14 [MeowSynth-08Zfghor] - bitbridge support
22:49:14 [MeowSynth-08Zfghor] - CLAP support
22:49:14 [MeowSynth-08Zfghor] - VST3 support
22:49:14 [MeowSynth-08Zfghor] 
22:49:14 [MeowSynth-08Zfghor] [Wine STDERR] /home/ldas/.local/share/yabridge/yabridge-host-32.exe: 36: exec: wine: not found
22:49:14 [MeowSynth-08Zfghor] The Wine host process has exited unexpectedly. Check the output above for more information.

### Anything else?

here is the .exe file:

```
ldas@framevirgin:~$ cat .local/share/yabridge/yabridge-host-32.exe
#!/bin/sh

appname="yabridge-host-32.exe.so"
# determine the application directory
appdir=''
case "$0" in
  */*)
    # $0 contains a path, use it
    appdir=`dirname "$0"`
    ;;
  *)
    # no directory in $0, search in PATH
    saved_ifs=$IFS
    IFS=:
    for d in $PATH
    do
      IFS=$saved_ifs
      if [ -x "$d/$appname" ]; then appdir="$d"; break; fi
    done
    ;;
esac

# figure out the full app path
if [ -n "$appdir" ]; then
    apppath="$appdir/$appname"
    WINEDLLPATH="$appdir:$WINEDLLPATH"
    export WINEDLLPATH
else
    apppath="$appname"
fi

# determine the WINELOADER
if [ ! -x "$WINELOADER" ]; then WINELOADER="wine"; fi

# and try to start the app
exec "$WINELOADER" "$apppath" "$@"
ldas@framevirgin:~$ 
```

and here is the launcher:

```
ldas@framevirgin:~$ cat ~/.local/share/applications/cockos-reaper.desktop 
[Desktop Entry]
Version=1.1
Type=Application
Name=REAPER
Comment=REAPER
Icon=cockos-reaper
Exec=reaper %F
MimeType=application/x-reaper-project;application/x-reaper-project-backup;application/x-reaper-theme;
Categories=Audio;AudioVideo;AudioVideoEditing;Recorder;Video;
StartupWMClass=REAPER
Name[en]=REAPER
Name[en_US]=REAPER
Name[en_US.UTF-8]=REAPER
Comment[en]=REAPER
Comment[en_US]=REAPER
Comment[en_US.UTF-8]=REAPER
Hidden=false
NoDisplay=false
StartupNotify=false
Terminal=false

ldas@framevirgin:~$ 
```

and finally here's my $PATH (idk if this is safe to post):

```
/home/ldas/.local/bin:/home/ldas/bin:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/power/x86/x86_energy_perf_policy:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/power/x86/turbostat:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/power/cpupower:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/perf:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/power/x86/x86_energy_perf_policy:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/power/x86/turbostat:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/power/cpupower:/usr/src/linux-headers-6.7.11-1-liquorix-amd64/tools/perf:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/opt/wine-staging/bin:/home/ldas/.local/share/yabridge
```

**Comment by aidanhadley** [comment]
yeah, the thing is, i have wine installed, and i can access it through `wine` on the terminal. the reason it's a 32-bit shared object file is because the plugin itself is a 32-bit plugin.

i'm now on arch, and have had no issue. after i finish this semester of school i'm going to tinker with it and see what's happening

**Comment by LeontKing2** [comment]
pretty strange... have you ran reaper with any arguments or such? (from terminal)

**Comment by aidanhadley** [comment]
no, i havent.

i'd like an update, i'm on fedora now and i'm having this issue. i had to add wine into my $PATH through `/opt/wine-staging/bin`. i feel like this is, perhaps, the issue. maybe running it from a launcher is not finding the exec?

**Comment by cbacott** [comment]
I am suddenly experiencing this same issue after upgrading to Debian Trixie. 

Under bookworm, wine was in the regular path, but now it's under /opt/wine-staging/ like these other distros. I did not have any issues under bookworm. I have made sure that the PATH is being globally set via /etc/profile.d/, and that the PATH is exported there. So, wine should definitely be available.

I can launch and load plugins without any problems when I use a terminal to launch reaper. I never get any errors with yabridge. But every time I use the launcher, it crashes when I load a plugin with the same issue. 

I am using yabridge 5.1.1 and Reaper 7.28.

I started testing with modifying the yabridge-host.exe file, since for my case that was the script being called rather than the 32-bit one. I finally got it to work correctly by setting the path just after line 33 like so:

```
# determine the WINELOADER
if [ ! -x "$WINELOADER" ]; then WINELOADER="wine"; fi

export PATH=$PATH:/opt/wine-staging/bin
```


And then it worked. I don't know what the difference here is between when it's launched by the launcher vs in a regular bash shell, environment wise, however it is definitely not getting the path from the global or local profile. I don't know what to propose to fix.

Of note, when I run from the launcher, this is the output from yabridge, a snip of it that would be relevant:
```
12:02:51 [Ampeg SVTVR Classic-4ABDRWko] wine prefix:   '<default>'
12:02:51 [Ampeg SVTVR Classic-4ABDRWko] wine version:  '<NOT FOUND>'
12:02:51 [Ampeg SVTVR Classic-4ABDRWko] 
12:02:51 [Ampeg SVTVR Classic-4ABDRWko] config from:   '<defaults>'
12:02:51 [Ampeg SVTVR Classic-4ABDRWko] hosting mode:  'individually, 64-bit'
12:02:51 [Ampeg SVTVR Classic-4ABDRWko] other options: '<none>'
```


For me, it's not finding wine to get the version at that point in execution. 
It does work, since I am setting the path explicitly in that script afterwards. I suppose this could be a temp workaround for users experiencing this issue.

**Comment by ishaanparlikar** [comment]
Hey @aidanhadley Did you figure it out? I am experiencing same issue.

**Comment by aidanhadley** [comment]
i have since moved to arch linux, and im not sure what i did to fix it on ubuntu, if i even did
